solr/connection.go

```diff
@@ -46,7 +49,24 @@ func HTTPPost(path string, data *[]byte, headers [][]string, username, password
 
 // HTTPGet make a GET request to url, headers are optional
 func HTTPGet(url string, headers [][]string, username, password string) ([]byte, error) {
-	client := &http.Client{Transport: &transport}
+	clientTimeout := 10 * time.Second
+	transport := &http.Transport{
```

Comment:
I think we can create one transport (and one client for all goroutines) so this feel a little bit unnecessary. And why it is only needed for get and not post

Can you share what problem you are solving? maybe we can find another way to solve the problem. As this implementation it will lock this configuration for everyone. If we want to control the transport we could instead provide a set transport function to give other a chance to control it as they want.

Reply:
sorry about my broken sentences, I re-phased

Reply:
The problem is the long default timeout from the transport. But I think your idea might be even better to let the user set the transport.

Reply:
that's good. Create a (suggested) SetTransport(* http. Transport) then anyone can control the network configurations as they want.

It is even better to reuse the client but that is probably something we do later.

Reply:
I think setting transport should do. I will do another PR if get the time ;)